cfgrammar/src/lib/yacc/parser.rs

```diff
@@ -1862,14 +1860,15 @@ x"
             "%start A
              %%
              A -> (): 'a1';
-             B -> (): 'b';
+             B -> (): 'bb';
              A -> (): 'a2';
-             C -> (): 'c';
+             C -> (): 'cc';
              A -> (): 'a3';
             ",
             YaccParserErrorKind::DuplicateRule,
             "A",
-            2
+            Span::new(38, 39),
+            vec![Span::new(94, 95), Span::new(150, 151)]
```

Comment:
IMO this macro has gotten a bit ugly now, it may just be better to add `Eq` to `YaccParserErrorKind`,
and then pass `YaccParserErrorKind::DuplicateRule([...])`, It means dropping the

"expected DuplicateRule(vec![])", on test failure, i'm not sure it is worth the complexity it adds by splitting these apart...

Reply:
I had to add a trait, to get the spans out of a kind, for testing against the expected string `"A"`,
I just added it private to the testsuite module, but was able to keep everything.

Reply:
I wonder if we should simplify this even further? Ultimately wouldn't something super simple like this work?

```rust
let r = yacc_kind, "%start A ...");
assert_eq!(r,
  YaccParserErrorKind::DuplicateRule { span: Span::new(...), kind: DuplicateRule(vec![Span::new(...), Span::new(...)]});
```